app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Api
+  module V1
+    class UsersController < ApiController
+      helper_method :user
+
+      def show; end
+
+      def profile
+        render :show
+      end
+
+      def update
+        user.update!(user_params)
+        render :show
+      end
+
+      private
+
+      def user_params
+        params.require(:user).permit(:email, :name, :gender)
+      end
+
+      def user
+        @user ||= user_id.present? ? User.find(user_id) : current_user
```

Comment:
Be careful with this helper method, this way you are fetching a user given the id present in the url, for example think about what would happen if somebody try to update other users info. This will be possible by changing the id in the request.